### PR TITLE
Improve auth deprecations

### PR DESCRIPTION
**Pull request by elia**
## Summary

Use the `deprecate` rails helper taking in method names to have a more standard message and improve the wording and suggested alternatives.

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [ ] I have written a thorough PR description.
- [ ] I have kept my commits small and atomic.
- [ ] I have used clear, explanatory commit messages.

The following are not always needed (~cross them out~ if they are not):

- [ ] I have added automated tests to cover my changes.
- [ ] I have attached screenshots to demo visual changes.
- [ ] I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- [ ] I have updated the README to account for my changes.

### CODE REVIEW

File: core/lib/spree/core/controller_helpers/auth.rb [hunk 1]
```diff
@@ -83,7 +72,11 @@ def try_spree_current_user
           end
         end
 
-        deprecate try_spree_current_user: :spree_current_user, deprecator: Spree::Deprecation
+        deprecate \
+          try_spree_current_user: :spree_current_user,
+          redirect_back_or_default: 'Please use `redirect_to stored_spree_user_location_or(...)` when using solidus_auth_devise.',
```
Comment (waiting-for-dev) [inline core/lib/spree/core/controller_helpers/auth.rb:77]: We're losing the information when not using solidus_auth_devise, right?

Comment (elia) [reply core/lib/spree/core/controller_helpers/auth.rb:77]: That information was bad, we wrote it (me and @cpfergus1) before having a clear picture of all the ramifications.

I think not having auth-devise in the store is really niche and probably something that required custom implementations anyway. What do you think about having an example implementation in the upgrading instructions?

Comment (waiting-for-dev) [reply core/lib/spree/core/controller_helpers/auth.rb:77]: That would be great, but I'm not very familiar with it. How would it look?

**Comment by kennyadsl** [comment]
@elia should we close this PR now that https://github.com/solidusio/solidus/pull/4856 has been merged?

**Pull request closed without merging**